### DISCUSSION TITLE
USB CDC Initialization issue

### DISCUSSION THREAD

**Discussion by a-domenick** [zephyrproject-rtos/zephyr#52092] [Q&A]
I have a setup that uses the uart-over-usb feature. I'm running into an issue that causes the system to infinitely send (garbage) data over the console upon booting if the following two conditions are true:

- The USB CDC ACM ring buffer fills up 
- There is a serial terminal open on my computer that is waiting for a COM port to open to initiate a connection to the device (e.g. when the device is plugged into USB) 

If the serial terminal opens the connection to the COM port some time after the COM port is available after getting plugged in, everything works ok. I think there's some issue that's causing the ring buffer to continuously print but I'm not sure how to go about debugging. 

Is there some known delay time that must be respected before trying to initiate a connection with the device?

**Comment by mkaranki** [comment]
I must admit I don't understand the overall system here, but could this be related to common case that host is echoing characters back until configured to not to do so, and your application is echoing them back as well? See https://michael.stapelberg.ch/posts/2021-04-27-linux-usb-virtual-serial-cdc-acm/

If that's the case, then a solution could be to wait DTR signal before sending anything to the host.

I was seeing garbage generated with Zephyr shell on CDC ACM. Turning on CONFIG_SHELL_BACKEND_SERIAL_CHECK_DTR which instructs the shell to wait DTR solved the issue.
